.github/workflows/test.yml

```diff
@@ -0,0 +1,51 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    name: "rake test"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ env.GITHUB_WORKSPACE }}/go/src/github.com/DataDog/agent-payload
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        path: go/src/github.com/DataDog/agent-payload
+    - name: "Install ruby"
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: head
+    - name: Install go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: "/home/runner/work/agent-payload/agent-payload/go/src/github.com/DataDog/agent-payload/go.mod"
+    - run: rake test
+      working-directory: "/home/runner/work/agent-payload/agent-payload/go/src/github.com/DataDog/agent-payload"
+      env:
+        GOPATH: "/home/runner/work/agent-payload/agent-payload/go"
+
+  codegen:
+    name: "rake codegen"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ env.GITHUB_WORKSPACE }}/go/src/github.com/DataDog/agent-payload
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        path: go/src/github.com/DataDog/agent-payload
+    - name: "Install ruby"
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: head
+    - name: Install go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: "/home/runner/work/agent-payload/agent-payload/go/src/github.com/DataDog/agent-payload/go.mod"
+    - run: rake codegen
+      working-directory: "/home/runner/work/agent-payload/agent-payload/go/src/github.com/DataDog/agent-payload"
+      env:
+        GOPATH: "/home/runner/work/agent-payload/agent-payload/go"
+    - name: Check for diffs
+      run: git diff --exit-code
+      working-directory: "/home/runner/work/agent-payload/agent-payload/go/src/github.com/DataDog/agent-payload"
```

Comment:
If we have a jobs default working dir, why do we need to specify it multiple times?

Reply:
Good point!